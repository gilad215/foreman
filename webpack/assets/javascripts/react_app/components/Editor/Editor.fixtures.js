import { noop } from '../../common/helpers';

export const editor = {
  name: 'editor',
  diffViewType: 'split',
  value: 'value',
  hosts: { host1: 1, host2: 2 },
  renderedValue: '',
  mode: 'Ruby',
  theme: 'Monokai',
  keyBinding: 'Default',
  selectedView: 'input',
  editorName: 'editor',
  isMaximized: false,
  isMasked: false,
  isRendering: false,
  readOnly: false,
  showError: true,
  errorText: '',
};

export const dropdowns = {
  modes: ['Text', 'Json', 'Ruby', 'Html_ruby', 'Sh', 'Xml', 'Yaml'],
  keyBindings: ['Default', 'Emacs', 'Vim'],
  themes: ['Github', 'Monokai'],
};

export const showBooleans = {
  showPreview: true,
  showImport: true,
  showHide: true,
};

export const editorOptions = {
  ...editor,
  ...dropdowns,
  data: {
    ...showBooleans,
    name: 'editor',
    title: 'title',
    template: '<? />',
    options: dropdowns,
    hosts: [
      { host: { name: 'host1', id: 1 } },
      { host: { name: 'host2', id: 2 } },
    ],
  },
  initializeEditor: noop,
  importFile: noop,
  revertChanges: noop,
  previewTemplate: noop,
  toggleModal: noop,
  changeDiffViewType: noop,
  changeEditorValue: noop,
  dismissErrorToast: noop,
  changeTab: noop,
  toggleMaskValue: noop,
  changeSetting: noop,
  toggleRenderView: noop,
  template: '<? />',
  renderPath: '/render/path',
};

export const serverRenderResponse = {
  data: '< renderedData />',
};

export const hosts = [
  { host: { name: 'host1', id: 1 } },
  { host: { name: 'host2', id: 2 } },
];
