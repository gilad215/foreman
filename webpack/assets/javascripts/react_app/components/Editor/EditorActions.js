import API from '../../API';
import { translate } from '../../common/I18n';

import {
  EDITOR_INITIALIZE,
  EDITOR_CHANGE_STATE,
  EDITOR_IMPORT_FILE,
  EDITOR_REVERT_CHANGES,
  EDITOR_EXEC_PREVIEW,
  EDITOR_MODAL_TOGGLE,
} from './EditorConstants';

export const initializeEditor = props => dispatch => {
  const {
    data: { template, locked, type, hosts },
    readOnly,
    isMasked,
    selectedView,
    isRendering,
    renderedValue,
    showError,
  } = props;

  const initialState = {};

  // initialize after changing editors
  initialState.value = template || '';
  if (readOnly !== locked) {
    if (locked === true) initialState.readOnly = true;
    else initialState.readOnly = false;
  }
  if (isMasked && type === 'templates') initialState.isMasked = false;
  if (selectedView !== 'input') initialState.selectedView = 'input';
  if (isRendering) initialState.isRendering = false;
  if (renderedValue !== '') initialState.renderedValue = '';
  if (showError) initialState.showError = false;

  if (hosts && hosts.length !== 0) {
    initialState.hosts = hosts.map(element => ({
      label: element.host.name,
      value: element.host.id,
    }));
    console.log(initialState.hosts);
  }

  dispatch({
    type: EDITOR_INITIALIZE,
    payload: { initialState },
  });
};

export const changeState = newState => (dispatch, getState) => {
  if (
    // key exists in editor state
    Object.prototype.hasOwnProperty.call(
      getState().editor,
      Object.keys(newState)[0]
    )
  ) {
    dispatch({
      type: EDITOR_CHANGE_STATE,
      payload: {
        newState,
      },
    });
  }
};

export const importFile = e => dispatch => {
  const reader = new FileReader();
  reader.onload = event => {
    dispatch({
      type: EDITOR_IMPORT_FILE,
      payload: {
        value: event.target.result,
      },
    });
  };
  reader.readAsText(e.target.files[0]);
};

export const revertChanges = template => dispatch => {
  dispatch({
    type: EDITOR_REVERT_CHANGES,
    payload: {
      value: template || '',
    },
  });
};

export const previewTemplate = (host, template, url) => dispatch => {
  const params = {
    template,
  };
  if (host != null) {
    /* eslint-disable camelcase */
    params.preview_host_id = host;
  }
  dispatch(changeState({ renderedValue: translate('Rendering...') }));

  API.post(url, params)
    .then(response => {
      dispatch({
        type: EDITOR_EXEC_PREVIEW,
        payload: {
          value: response.data,
        },
      });
    })
    .catch(error => {
      const newState = {
        errorText: translate(error.response.data),
        showError: true,
        renderedValue: translate(
          'Error during rendering, Return to Editor tab.'
        ),
      };
      dispatch({
        type: EDITOR_CHANGE_STATE,
        payload: {
          newState,
        },
      });
    });
};

export const toggleModal = () => ({
  type: EDITOR_MODAL_TOGGLE,
});
