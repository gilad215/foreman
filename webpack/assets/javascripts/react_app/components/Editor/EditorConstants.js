export const EDITOR_INITIALIZE = 'EDITOR_INITIALIZE';
export const EDITOR_CHANGE_STATE = 'EDITOR_CHANGE_STATE';
export const EDITOR_IMPORT_FILE = 'EDITOR_IMPORT_FILE';
export const EDITOR_REVERT_CHANGES = 'EDITOR_REVERT_CHANGES';
export const EDITOR_EXEC_PREVIEW = 'EDITOR_EXEC_PREVIEW';
export const EDITOR_MODAL_TOGGLE = 'EDITOR_MODAL_TOGGLE';
