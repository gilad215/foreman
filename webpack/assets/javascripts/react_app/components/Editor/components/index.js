import { Loadable } from '../../../utils';

export default Loadable({
  loader: () => import('./Editor'),
});
