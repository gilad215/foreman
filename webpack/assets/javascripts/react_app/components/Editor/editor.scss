#editor-dropdowns {
  float: right;

  #mode-dropdown, #keybindings-dropdown, #themes-dropdown {
    min-width: 85px;
    margin-right: 5px;
  }

  #editor-dropdowns {
    min-width: 85px;
    text-align: center;
  }
}

.diff-button {
  width: 60px;
  margin-right: 2px;
}

#fullscreen-btn, #import-btn {
  width: 40px !important;
}

.ace_editor_form {
  position: relative !important;
  height: 360px !important;
  width: 100% !important;
}

.ace_editor_modal {
  width: 100% !important;
  height: -webkit-fill-available !important;
}

.editor-modal {
  .modal-content { 
    height: -webkit-fill-available;
  }
  .modal-dialog {
    padding: 15px;
    width: auto !important;

    .input.modal-body {
      background-color: #363636;
    }

    .modal-body {
      padding: 0;

      .ace_editor {
        border: none;
      }

      #diff-container {
        height: 100%;

        #diff-table {
          max-height: none;
          border: none;
        }
      }
    }

    .input.modal-header.monokai {
      background-color: #363636;

      .pficon-close {
        color: #fff;
      }
    }
  }
}