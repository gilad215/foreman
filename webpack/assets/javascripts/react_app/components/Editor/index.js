import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as actions from './EditorActions';
import reducer from './EditorReducer';

import Editor from './components';

// map state to props
const mapStateToProps = ({ editor }) => ({
  value: editor.value,
  hosts: editor.hosts,
  renderedValue: editor.renderedValue,
  errorText: editor.errorText,
  mode: editor.mode,
  theme: editor.theme,
  keyBinding: editor.keyBinding,
  selectedView: editor.selectedView,
  editorName: editor.editorName,
  diffViewType: editor.diffViewType,
  isMaximized: editor.isMaximized,
  isMasked: editor.isMasked,
  isRendering: editor.isRendering,
  readOnly: editor.readOnly,
  showError: editor.showError,
});

// map action dispatchers to props
const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);

// export reducers
export const reducers = { editor: reducer };

// export connected component
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Editor);
