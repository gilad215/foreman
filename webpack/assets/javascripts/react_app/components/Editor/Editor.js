import React from 'react';
import PropTypes from 'prop-types';
import { ToastNotification } from 'patternfly-react';

import { noop } from '../../common/helpers';
import DiffView from '../DiffView/DiffView';
import EditorView from './components/EditorView';
import EditorNavbar from './components/EditorNavbar';
import EditorModal from './components/EditorModal';
import {
  EDITOR_THEMES,
  EDITOR_KEYBINDINGS,
  EDITOR_MODES,
} from './EditorConstants';
import './editor.scss';

class Editor extends React.Component {
  componentDidMount() {
    const {
      data: { hosts, locked, template, type },
      initializeEditor,
      isMasked,
      isRendering,
      readOnly,
      renderedValue,
      selectedView,
      showError,
    } = this.props;

    const initializeData = {
      hosts,
      isMasked,
      isRendering,
      locked,
      readOnly,
      renderedValue,
      selectedView,
      showError,
      template,
      type,
    };
    initializeEditor(initializeData);
  }

  render() {
    const {
      data: {
        name,
        renderPath,
        showHide,
        showImport,
        showPreview,
        template,
        title,
      },
      changeDiffViewType,
      changeEditorValue,
      changeSetting,
      changeTab,
      diffViewType,
      dismissErrorToast,
      editorName,
      errorText,
      hosts,
      importFile,
      isMasked,
      isMaximized,
      isRendering,
      keyBinding,
      mode,
      previewTemplate,
      readOnly,
      renderedValue,
      revertChanges,
      selectedView,
      showError,
      theme,
      toggleMaskValue,
      toggleModal,
      toggleRenderView,
      value,
    } = this.props;

    return (
      <div id="editor-container">
        <EditorNavbar
          changeDiffViewType={changeDiffViewType}
          changeTab={changeTab}
          changeSetting={changeSetting}
          modes={EDITOR_MODES}
          themes={EDITOR_THEMES}
          keyBindings={EDITOR_KEYBINDINGS}
          mode={isRendering ? 'Text' : mode}
          theme={theme}
          keyBinding={keyBinding}
          value={value}
          diffViewType={diffViewType}
          template={template}
          selectedView={selectedView}
          isDiff={template ? value !== template : false}
          isMasked={isMasked}
          isRendering={isRendering}
          importFile={importFile}
          showImport={showImport}
          showPreview={showPreview}
          showHide={showHide}
          revertChanges={revertChanges}
          previewTemplate={previewTemplate}
          hosts={hosts}
          renderPath={renderPath}
          toggleMaskValue={toggleMaskValue}
          toggleRenderView={toggleRenderView}
          toggleModal={toggleModal}
        />
        <ToastNotification
          id="preview_error_toast"
          type="error"
          className={showError ? '' : 'hidden'}
          onDismiss={() => dismissErrorToast()}
        >
          {errorText}
        </ToastNotification>
        <EditorView
          key="editorView"
          value={isRendering ? renderedValue : value}
          name={editorName}
          mode={isRendering ? 'Text' : mode}
          theme={theme}
          keyBinding={keyBinding}
          onChange={isRendering ? noop : changeEditorValue}
          readOnly={readOnly || isRendering}
          className={selectedView !== 'diff' ? 'ace_editor_form' : 'hidden'}
          isMasked={isMasked}
        />
        <div
          id="diff-table"
          className={selectedView === 'diff' ? '' : 'hidden'}
        >
          <DiffView
            oldText={template || ''}
            newText={value}
            viewType={diffViewType}
          />
        </div>
        <EditorModal
          key="editorModal"
          changeEditorValue={changeEditorValue}
          changeDiffViewType={changeDiffViewType}
          name={editorName}
          title={title}
          toggleModal={toggleModal}
          diffViewType={diffViewType}
          mode={isRendering ? 'Text' : mode}
          theme={theme}
          keyBinding={keyBinding}
          readOnly={readOnly}
          isMaximized={isMaximized}
          template={template || ''}
          editorValue={isRendering ? renderedValue : value}
          selectedView={selectedView}
          isMasked={isMasked}
        />
        <textarea className="hidden" name={name} value={value} readOnly />
      </div>
    );
  }
}

Editor.propTypes = {
  data: PropTypes.shape({
    showHide: PropTypes.bool,
    showImport: PropTypes.bool,
    showPreview: PropTypes.bool,
    template: PropTypes.string,
    name: PropTypes.string,
    title: PropTypes.string,
    renderPath: PropTypes.string,
  }).isRequired,
  changeDiffViewType: PropTypes.func.isRequired,
  changeEditorValue: PropTypes.func.isRequired,
  changeSetting: PropTypes.func.isRequired,
  changeTab: PropTypes.func.isRequired,
  diffViewType: PropTypes.string.isRequired,
  dismissErrorToast: PropTypes.func.isRequired,
  editorName: PropTypes.string.isRequired,
  errorText: PropTypes.string.isRequired,
  hosts: PropTypes.object.isRequired,
  importFile: PropTypes.func.isRequired,
  initializeEditor: PropTypes.func.isRequired,
  isMasked: PropTypes.bool.isRequired,
  isMaximized: PropTypes.bool.isRequired,
  isRendering: PropTypes.bool.isRequired,
  keyBinding: PropTypes.string.isRequired,
  mode: PropTypes.string.isRequired,
  previewTemplate: PropTypes.func.isRequired,
  readOnly: PropTypes.bool.isRequired,
  renderedValue: PropTypes.string.isRequired,
  revertChanges: PropTypes.func.isRequired,
  selectedView: PropTypes.string.isRequired,
  showError: PropTypes.bool.isRequired,
  theme: PropTypes.string.isRequired,
  toggleMaskValue: PropTypes.func.isRequired,
  toggleModal: PropTypes.func.isRequired,
  toggleRenderView: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};

export default Editor;
