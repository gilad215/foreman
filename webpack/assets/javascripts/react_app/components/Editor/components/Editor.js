import React from 'react';
import PropTypes from 'prop-types';
import { ToastNotification } from 'patternfly-react';

import { noop } from '../../../common/helpers';
import DiffView from '../../DiffView/DiffView';
import EditorView from './EditorView';
import EditorNavbar from './EditorNavbar';
import EditorModal from './EditorModal';
import '../editor.scss';

class Editor extends React.Component {
  componentDidMount() {
    const { initializeEditor } = this.props;
    initializeEditor(this.props);
  }

  render() {
    const {
      data: {
        showHide,
        showImport,
        showPreview,
        template,
        name,
        renderPath,
        options,
        title,
      },
      value,
      hosts,
      mode,
      theme,
      keyBinding,
      selectedView,
      renderedValue,
      editorName,
      diffViewType,
      changeState,
      importFile,
      revertChanges,
      previewTemplate,
      toggleModal,
      readOnly,
      showError,
      errorText,
      isMasked,
      isMaximized,
      isRendering,
    } = this.props;

    return (
      <div id="editor-container">
        <EditorNavbar
          value={value}
          diffViewType={diffViewType}
          template={template}
          mode={isRendering ? 'Text' : mode}
          theme={theme}
          keyBinding={keyBinding}
          selectedView={selectedView}
          modes={options.modes}
          themes={options.themes}
          keyBindings={options.keyBindings}
          changeState={changeState}
          toggleModal={toggleModal}
          showImport={showImport}
          showPreview={showPreview}
          showHide={showHide}
          importFile={importFile}
          revertChanges={revertChanges}
          previewTemplate={previewTemplate}
          hosts={hosts}
          renderPath={renderPath}
          isDiff={template ? value !== template : false}
          isMasked={isMasked}
          isRendering={isRendering}
        />
        <ToastNotification
          id="preview_error_toast"
          type="error"
          className={showError ? '' : 'hidden'}
          onDismiss={() => changeState({ showError: false })}
        >
          {errorText}
        </ToastNotification>
        <EditorView
          key="editorView"
          value={isRendering ? renderedValue : value}
          name={editorName}
          mode={isRendering ? 'Text' : mode}
          theme={theme}
          keyBinding={keyBinding}
          onChange={isRendering ? noop : changeState}
          readOnly={readOnly || isRendering}
          className={selectedView !== 'diff' ? 'ace_editor_form' : 'hidden'}
          isMasked={isMasked}
        />
        <div
          id="diff-table"
          className={selectedView === 'diff' ? '' : 'hidden'}
        >
          <DiffView
            oldText={template || ''}
            newText={value}
            viewType={diffViewType}
          />
        </div>
        <EditorModal
          key="editorModal"
          changeState={changeState}
          name={editorName}
          title={title}
          toggleModal={toggleModal}
          diffViewType={diffViewType}
          mode={isRendering ? 'Text' : mode}
          theme={theme}
          keyBinding={keyBinding}
          readOnly={readOnly}
          isMaximized={isMaximized}
          template={template || ''}
          editorValue={isRendering ? renderedValue : value}
          selectedView={selectedView}
          isMasked={isMasked}
        />
        <textarea className="hidden" name={name} value={value} readOnly />
      </div>
    );
  }
}

Editor.propTypes = {
  data: PropTypes.shape({
    showHide: PropTypes.bool,
    showImport: PropTypes.bool,
    showPreview: PropTypes.bool,
    template: PropTypes.string,
    name: PropTypes.string,
    title: PropTypes.string,
    renderPath: PropTypes.string,
    options: PropTypes.shape({
      modes: PropTypes.arrayOf(PropTypes.string),
      keyBindings: PropTypes.arrayOf(PropTypes.string),
      themes: PropTypes.arrayOf(PropTypes.string),
    }).isRequired,
  }).isRequired,
  changeState: PropTypes.func.isRequired,
  importFile: PropTypes.func.isRequired,
  revertChanges: PropTypes.func.isRequired,
  previewTemplate: PropTypes.func.isRequired,
  initializeEditor: PropTypes.func.isRequired,
  toggleModal: PropTypes.func.isRequired,
  renderedValue: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  hosts: PropTypes.object.isRequired,
  editorName: PropTypes.string.isRequired,
  diffViewType: PropTypes.string.isRequired,
  mode: PropTypes.string.isRequired,
  theme: PropTypes.string.isRequired,
  keyBinding: PropTypes.string.isRequired,
  selectedView: PropTypes.string.isRequired,
  readOnly: PropTypes.bool.isRequired,
  showError: PropTypes.bool.isRequired,
  errorText: PropTypes.string.isRequired,
  isMasked: PropTypes.bool.isRequired,
  isMaximized: PropTypes.bool.isRequired,
  isRendering: PropTypes.bool.isRequired,
};

export default Editor;
