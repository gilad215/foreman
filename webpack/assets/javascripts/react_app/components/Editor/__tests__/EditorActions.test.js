import API from '../../../API';
import { testActionSnapshotWithFixtures } from '../../../common/testHelpers';
import {
  initializeEditor,
  changeState,
  importFile,
  revertChanges,
  previewTemplate,
  toggleModal,
} from '../EditorActions';

import { editorOptions, serverRenderResponse } from '../Editor.fixtures';

jest.mock('../../../API');

const runRenderTemplate = serverMock => {
  API.post.mockImplementation(serverMock);

  return previewTemplate(11, '<template />', 'some/url');
};

const runChangeState = newState => dispatch => {
  const getState = () => ({
    editor: {
      value: '',
      renderedValue: '',
      mode: 'Ruby',
      theme: 'Monokai',
      keyBinding: 'Default',
      selectedView: 'input',
      editorName: 'editor',
      isMaximized: false,
      isMasked: false,
      isRendering: false,
      readOnly: false,
    },
  });
  changeState(newState)(dispatch, getState);
};

const e = { target: { files: [new File([new Blob()], 'filename')] } };

const fixtures = {
  'should initializeEditor': () =>
    initializeEditor({
      ...editorOptions,
      isMasked: true,
      selectedView: 'preview',
      isRendering: true,
      ...editorOptions.data,
      locked: true,
      type: 'templates',
      hosts: [
        { host: { id: 2, name: 'host2' } },
        { host: { id: 3, name: 'host3' } },
      ],
    }),

  'should initializeEditor unlocked': () =>
    initializeEditor({
      ...editorOptions,
      renderedValue: 'renderedValue',
      isMasked: true,
      selectedView: 'preview',
      isRendering: true,
      hosts: [
        { host: { id: 2, name: 'host2' } },
        { host: { id: 3, name: 'host3' } },
      ],
    }),

  'should importFile': () => importFile(e),

  'should toggleModal': () => toggleModal(),

  'should change mode to Html': () => runChangeState({ mode: 'Html' }),

  'should not change state': () => runChangeState({ notInState: 'notInState' }),

  'should revertChanges': () => revertChanges('<template />'),

  'should previewTemplate and succeed': () =>
    runRenderTemplate(async () => serverRenderResponse),
  'should previewTemplate and fail': () =>
    runRenderTemplate(async () => {
      throw new Error('some-error');
    }),
};

describe('Editor actions', () => testActionSnapshotWithFixtures(fixtures));
