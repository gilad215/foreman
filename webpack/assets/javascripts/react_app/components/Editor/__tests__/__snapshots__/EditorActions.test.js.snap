// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor actions should change mode to Html 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "newState": Object {
          "mode": "Html",
        },
      },
      "type": "EDITOR_CHANGE_STATE",
    },
  ],
]
`;

exports[`Editor actions should importFile 1`] = `Array []`;

exports[`Editor actions should initializeEditor 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "initialState": Object {
          "hosts": Object {
            "host2": 2,
            "host3": 3,
          },
          "isMasked": false,
          "isRendering": false,
          "readOnly": true,
          "selectedView": "input",
          "showError": false,
          "value": "<? />",
        },
      },
      "type": "EDITOR_INITIALIZE",
    },
  ],
]
`;

exports[`Editor actions should initializeEditor unlocked 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "initialState": Object {
          "hosts": Object {
            "host2": 2,
            "host3": 3,
          },
          "isRendering": false,
          "readOnly": false,
          "renderedValue": "",
          "selectedView": "input",
          "showError": false,
          "value": "",
        },
      },
      "type": "EDITOR_INITIALIZE",
    },
  ],
]
`;

exports[`Editor actions should not change state 1`] = `Array []`;

exports[`Editor actions should previewTemplate and fail 1`] = `
Array [
  Array [
    [Function],
  ],
]
`;

exports[`Editor actions should previewTemplate and succeed 1`] = `
Array [
  Array [
    [Function],
  ],
  Array [
    Object {
      "payload": Object {
        "value": "< renderedData />",
      },
      "type": "EDITOR_EXEC_PREVIEW",
    },
  ],
]
`;

exports[`Editor actions should revertChanges 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "value": "<template />",
      },
      "type": "EDITOR_REVERT_CHANGES",
    },
  ],
]
`;

exports[`Editor actions should toggleModal 1`] = `
Object {
  "type": "EDITOR_MODAL_TOGGLE",
}
`;
